hooks/pre-exit

```diff
@@ -74,7 +120,7 @@ upload() {
     curl_args+=("--form" "run_env[version]=$PLUGIN_VERSION")
   fi
 
-  curl_args+=("${BUILDKITE_PLUGIN_TEST_COLLECTOR_API_URL:-https://analytics-api.buildkite.com/v1/uploads}")
+  curl_args+=("${BUILDKITE_PLUGIN_TEST_COLLECTOR_API_URL:-https://analytics-api.buildkite.com/v1/uploads}" "-o" "${CURL_RESP_FILE}")
```

Comment:
If you only add it when the option is enabled you would also avoid having to update all existing tests :)